lib/dry/validation/schema.rb

```diff
@@ -48,7 +48,13 @@ def self.new(rules = config.rules, **options)
       def self.create_class(target, other = nil, &block)
         klass =
           if other
-            Class.new(other.class)
+            if other < Types::Struct
+              Validation.Schema(build: false) do
```

Comment:
this needs `target: target` otherwise schema's "path" won't be set, which we use to determine where nested structure is located in the input.